### PR TITLE
Fix/first quest

### PR DESCRIPTION
**Pull request by tikisailor**
fix FQ edge cases
- try catch
- captcha
- add /first-quest start command

### CODE REVIEW

File: src/app/commands/first-quest/FirstQuest.ts [hunk 1]
```diff
@@ -101,11 +120,50 @@ module.exports = class FirstQuest extends SlashCommand {
 		const { guildMember } = await ServiceUtils.getGuildAndMember(ctx);
 		let command: Promise<any>;
 		switch (ctx.subcommands[0]) {
+		case 'start':
```
Comment (nonsensecodes) [inline src/app/commands/first-quest/FirstQuest.ts:123]: @tikisailor, you can wrap the entire Switch statement in a `try/catch` block. That should help with readability, while grabbing any errors that filter through from each part of the switch.

File: src/app/events/GuildMemberAdd.ts [hunk 2]
```diff
@@ -32,48 +32,41 @@ export default class implements DiscordEvent {
...
+	});
+
+	captcha.on('timeout', () => {
+		member.kick('captcha timeout');
+	});
+
 	captcha.on('success', async () => {
 		await member.roles.add(fqConstants.FIRST_QUEST_ROLES.verified);
+
+		await new Promise(r => setTimeout(r, 1000));
```
Comment (nonsensecodes) [inline src/app/events/GuildMemberAdd.ts:65]: This is essentially a Promise within a Promise. You can remove the `await` here.

File: src/app/service/constants/constants.ts [hunk 3]
```diff
@@ -24,7 +24,8 @@ export default Object.freeze({
 	DB_COLLECTION_FIRST_QUEST_CONTENT: 'firstQuestContent',
 	DB_COLLECTION_FIRST_QUEST_TRACKER: 'firstQuestTracker',
 	DB_COLLECTION_FIRST_QUEST_POAPS: 'firstQuestPOAPs',
-	MONGODB_URI_PARTIAL: `${process.env.MONGODB_PREFIX}://${process.env.MONGODB_USERNAME}:${process.env.MONGODB_PASS}@${process.env.MONGODB_CLUSTER}/`,
+	MONGODB_URI_PARTIAL: `mongodb+srv://${process.env.MONGODB_USERNAME}:${process.env.MONGODB_PASS}@${process.env.MONGODB_CLUSTER}${process.env.MONGODB_URL_ENDPOINT}`,
```
Comment (nonsensecodes) [inline src/app/service/constants/constants.ts:27]: Please stop editing this. Check your `.env` file for the proper syntax.

**Review by nonsensecodes — CHANGES_REQUESTED** [review]
A few remarks for your review, @tikisailor 🙏

File: src/app/service/constants/constants.ts (hunk 3 above)
Comment (tikisailor) [reply src/app/service/constants/constants.ts:27]: oops

File: src/app/events/GuildMemberAdd.ts (hunk 2 above)
Comment (tikisailor) [reply src/app/events/GuildMemberAdd.ts:65]: It doesn't seem to work without await.

Comment (tikisailor) [reply src/app/events/GuildMemberAdd.ts:65]: It doesn't seem to work without await.

File: src/app/commands/first-quest/FirstQuest.ts (hunk 1 above)
Comment (tikisailor) [reply src/app/commands/first-quest/FirstQuest.ts]: done

**Comment by amaredeus** [comment]
Closing this as another PR with merged changes is raised https://github.com/BanklessDAO/degen-bankless-dao/pull/18

**Pull request closed without merging**